### ISSUE TITLE
Floating point roundoff on Windows

### ISSUE THREAD

**Issue by lswaenen** [pyinstaller/pyinstaller#2188]
Hi all,

Executing the following snippet 

```
#main.py
import math

a = 0.999999999999999888977697537484
print("%.30f" % a)
print("%.30f" % math.sqrt(a))
```

on Windows (32 bit) as `python main.py` outputs
0.999999999999999888977697537484
0.999999999999999888977697537484

Executing main.exe after `pyinstaller main.py`, however, yields
0.999999999999999888977697537484
1.000000000000000000000000000000

which means the .exe rounds this result incorrectly.
On Linux, this issue does not occur.
PyInstaller version is the newest at time of writing (3.3dev0)

Kind regards,
Lense

**Comment by htgoebel** [comment]
I can't tell why this is happening here, but PyInstaller just uses the Python interpreter.

Please try running `python pyinstaller.py main.py` to ensure you are using *exaclty* the same python interpreter (and paths) in both test-cases.

**Comment by lswaenen** [comment]
@htgoebel I tried what you suggested, but no difference.
Below is a copy/paste of the pyinstaller output

> 225 INFO: PyInstaller: 3.3.dev0+b78bfe5
225 INFO: Python: 2.7.12
236 INFO: Platform: Windows-7-6.1.7601-SP1
236 INFO: wrote Z:\Shape control\project_files\py2exe_bug.spec
246 INFO: UPX is not available.
256 INFO: Extending PYTHONPATH with paths
['Z:\\Shape control\\project_files\\src', 'Z:\\Shape control\\project_files']
256 INFO: checking Analysis
256 INFO: Building Analysis because out00-Analysis.toc is non existent
256 INFO: Initializing module dependency graph...
276 INFO: Initializing module graph hooks...
492 INFO: running Analysis out00-Analysis.toc
492 INFO: Adding Microsoft.VC90.CRT to dependent assemblies of final executable
  required by C:\Users\Lense\Anaconda2\envs\noPyQtClone\python.exe
677 INFO: Found C:\Windows\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.4940_none_f47ed0f6f6564d90.manifest
809 INFO: Searching for assembly x86_Microsoft.VC90.CRT_1fc8b3b9a1e18e3b_9.0.30729.4940_none ...
809 INFO: Found manifest C:\Windows\WinSxS\Manifests\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.4940_none_50916076bcb9a742.manifest
809 INFO: Searching for file msvcr90.dll
809 INFO: Found file C:\Windows\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.4940_none_50916076bcb9a742\msvcr90.dll
820 INFO: Searching for file msvcp90.dll
820 INFO: Found file C:\Windows\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.4940_none_50916076bcb9a742\msvcp90.dll
820 INFO: Searching for file msvcm90.dll
820 INFO: Found file C:\Windows\WinSxS\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.4940_none_50916076bcb9a742\msvcm90.dll
944 INFO: Found C:\Windows\WinSxS\Manifests\x86_policy.9.0.microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.4940_none_f47ed0f6f6564d90.manifest
944 INFO: Adding redirect Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 4940)
1180 INFO: Caching module hooks...
1190 INFO: Analyzing Z:\Shape control\project_files\src\py2exe_bug.py
1200 INFO: Loading module hooks...
1200 INFO: Loading module hook "hook-encodings.py"...
5940 INFO: Looking for ctypes DLLs
5950 INFO: Analyzing run-time hooks ...
5960 INFO: Looking for dynamic libraries
6309 INFO: Looking for eggs
6309 INFO: Using Python library C:\Users\Lense\Anaconda2\envs\noPyQtClone\python27.dll
6309 INFO: Found binding redirects: 
[BindingRedirect(name=u'Microsoft.VC90.CRT', language=None, arch=u'x86', oldVersion=(9, 0, 21022, 8), newVersion=(9, 0, 30729, 4940), publicKeyToken=u'1fc8b3b9a1e18e3b')]
6320 INFO: Warnings written to Z:\Shape control\project_files\build\py2exe_bug\warnpy2exe_bug.txt
6463 INFO: checking PYZ
6463 INFO: Building PYZ because out00-PYZ.toc is non existent
6463 INFO: Building PYZ (ZlibArchive) Z:\Shape control\project_files\build\py2exe_bug\out00-PYZ.pyz
7325 INFO: checking PKG
7325 INFO: Building PKG because out00-PKG.toc is non existent
7325 INFO: Building PKG (CArchive) out00-PKG.pkg
7480 INFO: Bootloader C:\Users\Lense\Anaconda2\envs\noPyQtClone\lib\site-packages\PyInstaller\bootloader\Windows-32bit\run.exe
7480 INFO: checking EXE
7480 INFO: Building EXE because out00-EXE.toc is non existent
7480 INFO: Building EXE from out00-EXE.toc
7480 INFO: Appending archive to EXE Z:\Shape control\project_files\build\py2exe_bug\py2exe_bug.exe
7674 INFO: checking COLLECT
7674 INFO: Building COLLECT because out00-COLLECT.toc is non existent
7674 INFO: Removing dir Z:\Shape control\project_files\dist\py2exe_bug
7674 INFO: Building COLLECT out00-COLLECT.toc
7736 INFO: Redirecting Microsoft.VC90.CRT version (9, 0, 21022, 8) -> (9, 0, 30729, 4940)

Can anyone else confirm/reproduce this behavior?

**Comment by dhyams** [comment]
Maybe I'm being a little obtuse, but you're trying to print *30 digits* of a floating point number.  For a double precision, you have at most 16 digits after the decimal that are significant (you have a 52 bit mantissa).  Printing more than this will just give garbage, as far as I know.  The fact that the garbage is different in each case, I don't think is significant.  And I would use an %lf to print, not %f, even so.

I'm a little shocked that even directly printing that number after assigning so much precision to it, even prints the same number back.  That's why I think I must be missing something.  And isn't it suspicious that "a" and "sqrt(a)" are exactly the same number?  Doesn't that imply the sqrt(a) is not doing anything?  a==sqrt(a) only for 1.0.

For what it's worth, I did run the sample code in this issue both on a Mac and Windows (64 bit with 64 bit python), and I got the .9999999999.... number in all situations; both running directly with python and after pyinstaller'ed.

I could close this bug.
